test/unit/space_index_test.lua

```diff
@@ -348,8 +348,38 @@ function g.test_not_stop_after_drop_and_recreate(cg)
     end)
 
     helpers.iteration_result = {}
+
+    -- This should be atomic to prevent yielding into the expirationd task with
+    -- partially applied changes (when the old space is dropped but the new one
+    -- is not created yet).
+    --
+    -- The space drop is transactional if the tarantool/taratool#8751 is merged,
+    -- so it yields on commit. After that point the space is dropped.
+    --
+    -- The expirationd task yields on each tuple drop if the task does not have
+    -- the atomic_teration option set.
+    --
+    -- So without this begin/commit guard the following situation is possible:
+    -- 1. The expirationd task has started.
+    -- 2. The task has removes one tuple, which causes yield.
+    -- 3. The current fiber drops the space and yields to the expirationd task.
+    -- 4. The expirationd task finishes the iteration and checks if the space
+    --    is vynil to update the vinyl_assumed_space_len, but the space does
+    --    not exist anymore, so it fails causing the task restart.
+
+    -- Since the test is also executed against old versions of Tarantool, let's
+    -- only wrap the space drop and recreation into transaction on the versions
+    -- that support MVCC-safe transactional DDL operation subset (e. g. versions
+    -- after tarantool/tarantool@8f4be3227635ef606581dda7e85f254a96347c0e).
```

Comment:
I think this comment is wrong. And there is no any relation to MVCC here. Transactional DDL is available since 2.2.1 (https://github.com/tarantool/tarantool/commit/f266559b167b4643c377724eebde9651877d6cc9)

Reply:
Yes, looks like the the commit fixed the bug that can't be reproduced using regular `box.begin`/`box.commit` in lua.

Updated the version.